src/communication/NetworkClient.ts

```diff
@@ -161,28 +145,76 @@ export default class NetworkClient {
     return embedded[binderName] as R[];
   }
 
-  iterate<R>(relativePath: string, type: string, query: Record<string, any> = {}): HelpfulIterator<R> {
+  iterate<R>(pathname: string, binderName: string, query: Maybe<SearchParameters>, valuesPerMinute = 500): HelpfulIterator<R> {
```

Comment:
2: `query` was optional, because the `iterate` methods in the binders used to kind-of just forward their arguments into `NetworkClient.iterate`. Now they extract `valuesPerMinute` anyway, so there will always be a `query`. (Come to think of it, it doesn't even have to be `Maybe`d any more.)

1: The `500` is not based on anything. The good news is that we can always change these values when we get new insights, it wouldn't cause a breaking change. The bad news is that we have little data to go by.

There are several instances where someone made a bunch of requests to list endpoints (I don't have a number more specific than "a bunch") and caused a spike, making the Mollie API less responsive. That's what we're trying to avoid. I would guess that requesting, say, 2000 values in a minute should be fine.

I actually dislike the "per minute" part as well. A minute feels long. Maybe we should throttle in multiples of 20 seconds instead of minutes, but still advertise it as "per minute" (and divide the `valuesPerMinute` by three)?

Reply:
2: I see. in that case yes, I agree it shouldn't be `Maybe`

1: Agree on setting a higher default value, although I still think this would be best defined on "hard evidence".
The term `valuesPerMinute` is fine and I don't think it's confusing, even if the throttled chunks come in with less time apart, which I also think its a good idea. 100 values every 10 seconds would still be 600 `valuesPerMinute` and it's a lot more intuitive than 100 `valuesEvery10Seconds`.

Reply:
I just changed the default to 5000 values per minute. Additionally, throttling now happens in increments of six seconds instead of minutes. As a result, the throttling now defaults to 500 values per six seconds in practice.